pkg/controllers/dashboard/controller.go

```diff
@@ -47,6 +50,10 @@ func Register(ctx context.Context, wrangler *wrangler.Context, embedded bool, re
 
 	clusterconnected.Register(ctx, wrangler)
 
+	if features.UIPlugin.Enabled() {
+		plugin.Register(ctx, namespace.UIPluginNamespace, wrangler.Catalog.UIPlugin(), wrangler.Catalog.UIPlugin().Cache())
```

Comment:
The functionality fails in HA mode check the video pasted. Every 3rd call only passes since this controller is only started in the leader pod and not on the other 2 slave pods. 

We need another wrangler handler like https://github.com/rancher/rancher/blob/release/v2.9/pkg/controllers/dashboard/helm/repo.go#L63 for followers.

Reply:
Good catch. As discussed, the controllers are now initialized in all rancher pods and not just the lead